### PR TITLE
feat: new event to send emails to winners when the ranking finishes and fix in the organization of events

### PR DESCRIPTION
**Pull request by PatricioPoncini**
### `Changes`

- new event to send emails to winners when the ranking finishes
- fix in the organization of events

### `docs`

docs are in the respective files

### CODE REVIEW

**Comment by PatricioPoncini** [comment]
**Ticket:** https://legendaryum.atlassian.net/jira/software/projects/LE/boards/1?assignee=712020%3A71750ee7-8648-475c-99c7-c39c60a8e3fc&selectedIssue=LE-2583

The idea of the event is to pass something like this to the microservice 'legend-send-email' so that it can create a template with the data such as title and rewardType for each object, and send it to each of the winners with their respective reward
ex.
```json
{
  "completedRankings": [
    {
      "title": "Top Achievers of the Month",
      "rewardType": "Legends",
      "winners": [
        {
          "userId": "507f1f77bcf86cd799439012",
          "reward": 500
        },
        {
          "userId": "507f1f77bcf86cd799439011",
          "reward": 300
        }
      ]
    },
    {
      "title": "Weekly Challenge Winners",
      "rewardType": "CodeExchange",
      "winners": [
        {
          "userId": "507f1f77bcf86cd799439013",
          "reward": 574745
        },
        {
          "userId": "507f1f77bcf86cd799439014",
          "reward": 389227
        }
      ]
    }
  ]
}
```

File: packages/legend-transac/src/@types/event/events.ts [hunk 1]
```diff
@@ -199,6 +235,7 @@ export const microserviceEvent = {
     'COINS.UPDATE_SUBSCRIPTION': 'coins.update_subscription',
     'LEGEND_MISSIONS.COMPLETED_MISSION_REWARD': 'legend_missions.completed_mission_reward',
     'LEGEND_MISSIONS.ONGOING_MISSION': 'legend_missions.ongoing_mission',
+    'LEGEND_RANKINGS.SEND_EMAIL_TO_WINNERS': 'legend_rankings.send_email_to_winners',
```
Comment (jym272) [inline packages/legend-transac/src/@types/event/events.ts:238]: In contrast to a saga command, which explicitly dictates the order of actions to a specific microservice, an event simply represents something that has occurred within a microservice, and subscribers react based on their interest. The event doesn't issue directives or control what actions should follow. Therefore, a more fitting name for the event would be 'LEGEND_RANKINGS.WINNERS_DETERMINED' or something similar.

File: packages/legend-transac/src/@types/event/events.ts [hunk 2]
```diff
@@ -43,6 +43,36 @@ export const PaymentEmailTypes = {
  */
 export type PaymentEmailType = (typeof PaymentEmailTypes)[keyof typeof PaymentEmailTypes];
 
+/**
+ * Types of Rankings Rewards Types
+ */
+export const RankingsRewardsTypes = {
+    LEGENDS: 'Legends',
```
Comment (jym272) [inline packages/legend-transac/src/@types/event/events.ts:50]: Take account this PascalCase in the other libraries Pr's.

**Review by jym272 — CHANGES_REQUESTED** [review]

File: packages/legend-transac/src/@types/event/events.ts (hunk 1 above)
Comment (PatricioPoncini) [reply packages/legend-transac/src/@types/event/events.ts]: Based on that definition, what is happening is that the rankings have finished, which is why I think the event name `legend_rankings.rankings_finished` is better

**Review by jym272 — APPROVED** [review]

**Pull request merged**